src/reference.rs

```diff
@@ -169,6 +169,27 @@ fn parse_url(input: &str) -> Result<Reference, Error> {
         .map(String::from)
         .collect();
 
+    // Figure out if the first component is a registry String, and assume the
+    // default registry if it's not.
+    let first = components
+        .pop_front()
+        .ok_or(Error::from("missing image name"))?;
+
+    let registry = if regex::Regex::new(r"(?x)
```

Comment:
Mmmh, maybe a `format!("^{hostname}{opt_port}$")` is a bit more readable instead of inline comments? Also, what's the leading `(?x)` for?

Reply:
Hehe, I had the format variant previously but didn't like it as much. If you feel strongly about it I'll revert back.

Reply:
>  what's the leading `(?x)` for?

http://www.rexegg.com/regex-modifiers.html#freespacing

Reply:
Ack, I'm fine with keeping it as is.